dev/NavigationView/NavigationView.cpp

```diff
@@ -1483,14 +1485,28 @@ void NavigationView::UpdatePaneLayout()
         const auto totalAvailableHeight = [this]() {
             if (const auto &paneContentRow = m_itemsContainerRow.get())
             {
-                // 20px is the padding between the two item lists
-                if (const auto &paneFooter = m_leftNavFooterContentBorder.get())
+                const double bottomMargin = [this]() {
+                    if (const auto itemsContainer = m_itemsContainer.get())
+                    {
+                        return itemsContainer.Margin().Bottom;
+                    }
+                    return 0.0;
+                }();
+
+                // We have 9px margin at the bottom, that's why we subtract 9 everywhere
+                if (m_menuItemsSource && m_footerItemsSource && m_menuItemsSource.Count() * m_footerItemsSource.Count() == 0 && !IsSettingsVisible())
                 {
-                    return paneContentRow.ActualHeight() - 29 - paneFooter.ActualHeight();
+                    // Either of these collections has no items, let it take all of the available space then.
+                    return paneContentRow.ActualHeight() - bottomMargin;
+                }
+                else if (const auto &paneFooter = m_leftNavFooterContentBorder.get())
```

Comment:
this if statement will always be true. I think we first have to see if PaneFooter() property is set. If it is - then we execute the clause. If there's no panefooter set by the user, why would we reserve 20 pixels?

Reply:
As well as bottom margin is not 9, it's 8?
![image](https://user-images.githubusercontent.com/21356912/113824522-c38fb200-9734-11eb-9b18-6da49bb4de0f.png)
The entire calculation of actual height here could be something like this:
![image](https://user-images.githubusercontent.com/21356912/113824597-d5715500-9734-11eb-88b9-587ec40c85ae.png)

Reply:
We're also missing PaneFooter() property changed. So if PaneFooter is dynamically set - it wouldn't properly update the layout

Reply:
That's a good point. It looks like that you already have the code for some of this. Would you like to push the changes first and I'll handle the rest after your push @beervoley ?

Reply:
@chingucoding yup, 5 min

Reply:
@chingucoding, I've updated the code and refactored this:
![image](https://user-images.githubusercontent.com/21356912/113919021-f4550300-9797-11eb-9afa-a39f769d12ac.png)
a little bit so it's easier to read. Please feel free to refactor it if it doesn't feel right to you :)

Reply:
Code looks good to me, thank you so much for helping out @beervoley!